10-angular/ui_router/readme.md

```diff
@@ -0,0 +1,389 @@
+
+This has the same end result as `git clone` except there's no `.git` folder in there. The reason we're asking you to use this ZIP file is because it gives you a nice little folder with its own version number: the subsequent folders you download will have their own numbers as well, so you won't have a zillion duplicate folders.
+
+Note, though, that since there's no `.git` folder that means you can't make commits and branch and track changes. If you want to be able to do that, take a moment right now to `git init` and make an initial commit.
+
+## Open the file in Atom and your browser
+
+It's all terribly exciting: the word `Grumblr`, and that's pretty much it.
+
+Grumblr is like Tumblr, only grumblier. It's a two-model CRUD app with posts and comments. Down the road it'll have users too.
+
+In the coming classes you're going to be interacting with data from an API that we provide. For this class, though, we'll just be hardcoding data and getting some views up and running.
+
+## Under the hood
+
+Open up your Chrome console and enter `console.log(angular)`. Click on the output to expand it.
+
+This is the `angular` object. It contains absolutely everything that has to do with Angular. You can see it has some weird things attached to it, like `getTestability` and `$$csp`, but it also has a whole bunch of pretty easy-to-read things: `angular.isDate`, `angular.forEach`, `angular.toJson`, and so on.
+
+Q. What are all these things attached to the `angular` object? Why are they here?
+---
+> They're convenience methods -- things that don't come built-in with Javascript, but are handy to have around.
+
+**Those of you using Chrome** click on `module`, then `<function scope>`. This is a Chrome-only thing that shows all of the variables available in the scope of this object. Click on `Closure`, then `b: Object`.
+
+> Those of you not using Chrome can just observe me doing this. What we're doing here doesn't have any practical application; we're just exploring.
+
+After a lot of hunting, you can see something interesting: These are all the **modules** available to us from the Angular libraries included in this file.
+
+![Inside module](http://i.imgur.com/1eLxSDz.png)
+
+## Modules
+
+Modules -- not to be confused with *models* -- are the building block of Angular. Every Angular app is a collection of modules interacting with each other. They're kind of like Javascript objects, and kind of like gems, and kind of like models, and kind of like jQuery plug-ins.
+
+If you click on any one of those modules, and then click on `requires`, you can see that each one requires another module. This is evidence of how they all work off of each other.
+
+**I'm going to remove two Angular libraries** out of the three included on this page, just by deleting their `<script>` tags. Now I only have two modules: `ng` and `ngLocale`. These are the only two that come out-of-the-box with Angular.
```

Comment:
aside: why are we using "ajax.googleapis.com" for angular and "cdnjs.cloudflare.com" for the others?